### DISCUSSION TITLE
How to add AZERTY (FR/BE) as eD layout?

### DISCUSSION THREAD

**Discussion by rzhlm** [DreymaR/BigBagKbdTrixPKL#53] [Q&A] [answered]
What way would be the easiest way be to add AZERTY (French & Belgian) layout, in order to also have the Extend features on those?

**Comment by DreymaR** [comment, marked as answer]
AZERTY is a bit special in that it changes several letter positions in the main block. Therefore, I'd make a new BaseLayout for it. In the QWERTY folder there's a QWERTY BaseLayout; copy and edit it accordingly.

**Reply by rzhlm** [reply]
I got it working, mostly:
All keys and dead keys work.
But normal shortcuts such as CTRL+S, CTRL+Z or CTRL+W don't work anymore or are doing other things. Where do I look to fix that?

**Reply by DreymaR** [reply]
You need to look to the VK codes. They are what governs shortcut behavior for the most part. So when you move a key, change its scan code so you move the whole key, not just some of its entries. Or edit it carefully to ensure everything gets right, including the VK code.

EPKL allows using `vc###` as a synonym for the VK that's usually on the `QW###` key. E.g., for the `QW_E` key, its VK will be `vc_E`. This is handy if you don't know the actual codes. So to remap, say, the F key to E like Colemak does you'll want the `QW_E` key to produce the `vc_F` VK code, and of course `f F` etc entries.
